_posts/2017-03-06-new-topology-system.md

```diff
@@ -0,0 +1,102 @@
+     1 2                       1 LYS
+     2 1                       2 ALA
+     3 1
+     4 2
+```
+
+calling `AtomGroup.resnames` for an `AtomGroup` with atoms [2, 0, 1, 2] will yield (pseudocode):
+
+```
+"Atoms->Residues"[[2, 0, 1, 2]] --> [1, 0, 2, 1]
+"Resnames"[[1, 0, 2, 1]]        --> ['LYS', 'GLU', 'ALA', 'LYS']
+```
+
+This scheme only works if each atom is a member of one and only one residue, and likewise if residues are members of one and only one segment.
+Furthermore, `AtomGroup`s, `ResidueGroup`s, and `SegmentGroup`s are very thin, storing only the indices of their members as a `numpy` array.
+This gives a number of advantages:
+
+1. **Performance**. We get up to an 8x speedup over the old scheme when accessing attributes. Setting attributes can give up to a 40x speedup.
+2. **Memory**. We don't store, for example, a resname for each atom, but instead store attributes at the level they make sense for.
+3. **Consistency**. Since attributes are stored in one place, we avoid cases where the topology is in an inconsistent state, e.g. two atoms in the same residue give a different resname.
+4. **No staleness**. Because e.g. `ResidueGroup`s are only an array of indices, not a list of `Residue` objects generated upon creation of the group, changes of resiude-level properties by another `ResidueGroup` are always reflected consistently by every other one. Data is not duplicated anywhere in this scheme, and is all contained in the `Topology` object.
+
+For further performance comparisons, check out this [notebook](http://nbviewer.jupyter.org/gist/dotsdl/0e0fbd409e3e102d0458).
+
+## External changes that may affect how you use MDAnalysis
+
+Previously, every object except ``Atom`` subclassed from ``AtomGroup``.
+This meant that calling `.positions` of would give you the positions of the ``Atom``s contained within that group.
+
+Previous class structure:
+```
+Atom
+
+AtomGroup  -> Residue
+           -> ResidueGroup -> Segment
+                           -> SegmentGroup
+```
+New class structure:
+```
+Group    -> AtomGroup
+         -> ResidueGroup
+         -> SegmentGroup
+
+Atom
+Residue
+Segment
+```
+
+Now each object only contains information pertaining to that particular object.
+A ``Residue`` object only yields information about the residue; to get to the atoms, use ``Residue.atoms``.
+
+### Why this was changed
+
+Previously everything inheriting from ``AtomGroup`` made it unclear at what level of topology a given method or attribute was working on.
+For example, does ``ResidueGroup.charges`` give the charge of the residues or the atoms?
+Also, it was unclear what size a given output would be (see [issue 411](https://github.com/MDAnalysis/mdanalysis/issues/411)).
+
+### How to work around this
+
+To access atom-level information from anything that isn't an ``AtomGroup``, use the `.atoms` level accessor.
+For example, changing all `.positions` calls on anything that isn't an `AtomGroup` to `.atoms.positions`.
```

Comment:
Yes it returns an attribute error now.  I was adding the remark because above @dotsdl mentions that this was allowed but not well defined in the old topology.  It may be worth mentioning that we have removed ill defined attributes like `ResidueGroup.positions`